Source/ThirdParty/ik/CMakeLists.txt

```diff
@@ -38,19 +34,20 @@ option (IK_MEMORY_DEBUGGING "Global switch for memory options. Keep track of the
 cmake_dependent_option (IK_MEMORY_BACKTRACE "Generate backtraces for every malloc(), making it easy to track down memory leaks" "${DEFAULT_MEMORY_DEBUGGING}" "IK_MEMORY_DEBUGGING AND NOT WEB" FALSE)
 
 # Enable restrict keyword in quaternion and vector operations if not in debug
-set (IK_RESTRICT "")
-if (NOT CMAKE_BUILD_TYPE STREQUAL Debug)
+if (NOT IK_RESTRICT)
+    set (IK_RESTRICT "")
```

Comment:
Of course, the "lol" snippet of yours won't work. The cache variable value must be "truthy" in order to not get into the protected block again. My previous snippet just want to give you the idea. I should probably use another more proper variable name instead of IK_RESTRICT. So let me try that again.

```
if (NOT IK_RESTRICT_CHECKED)
 message ("LOL once")
 set (IK_RESTRICT_CHECKED 1 CACHE INTERNAL "IK restrict check")
endif ()
```